wrangler-service/src/main/java/co/cask/wrangler/service/FileTypeDetector.java

```diff
@@ -41,19 +39,18 @@
 
   public FileTypeDetector() {
     this.extensions = new HashMap<>();
-    File file = new File(Explorer.class.getClassLoader().getResource("file.extensions").getFile());
-    try {
-      Scanner scanner = new Scanner(file);
-      while(scanner.hasNext()) {
+    try (InputStream is = getClass().getClassLoader().getResource("file.extensions").openStream()) {
+      Scanner scanner = new Scanner(is, "UTF-8");
+      while (scanner.hasNext()) {
         String line = scanner.nextLine();
-        String[] parts = line.split("[\t+\\s+]");
+        String[] parts = line.split("\\s+");
         if (parts.length == 2) {
           extensions.put(parts[0], parts[1]);
         }
       }
       scanner.close();
```

Comment:
The `scanner.close()` will close the `InputStream`. So you can actually do:

```java
try (Scanner scanner = new Scanner(getClass().getClassLoader().....)) {
}
```

Sorry didn't catch that earlier.